src/Message/ApplePayAuthorizeRequest.php

```diff
@@ -0,0 +1,443 @@
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('merchantIdentifier', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getDisplayName()
+    {
+        return $this->getParameter('displayName');
+    }
+
+    /**
+     * Sets the display name for the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setDisplayName($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('displayName', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getApplicationUrl()
+    {
+        /**
+         * @var null|string
+         */
+        return $this->getParameter('applicationUrl');
+    }
+
+    /**
+     * Sets the application url or the "initiativeContext" for the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setApplicationUrl($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('applicationUrl', $value);
+    }
+
+    /**
+     * Get HTTP Method.
+     *
+     * This is nearly always POST but can be over-ridden in sub classes.
+     *
+     * @return string
+     */
+    public function getHttpMethod()
+    {
+        return 'POST';
+    }
+
+    /**
+     * @return array
+     */
+    public function getData()
+    {
+        // We must have the validation URL set. If it isn't, don't make the call.
+        $this->validate('validationUrl');
+        $data = array();
+
+        // Required request parameters for Apple Session object.
+        $data['merchantIdentifier'] = $this->getMerchantIdentifier();
+        $data['displayName'] = $this->getDisplayName();
+        // Default parameter for Apple Pay on the Web.
+        $data['initiative'] = 'web';
+        $data['initiativeContext'] = $this->getApplicationUrl();
+
+        return $data;
+    }
+
+    /**
+     * @param       $data
+     * @param array $headers
+     *
+     * @return static
+     */
+    protected function createClientRequest($data, array $headers = null)
+    {
+        $pemCertPath = $this->getPemCertPath();
+        $keyCertPath = $this->getKeyCertPath();
+        $keyCertPassword = $this->getKeyCertPassword();
+
+        // Setting options for the request.
+        $config                            = $this->httpClient->getConfig();
+        $curlOptions                       = $config->get('curl.options');
+        $curlOptions[CURLOPT_SSLVERSION]   = 6;
+        $curlOptions[CURLOPT_SSLCERT]      = $pemCertPath;
+        $curlOptions[CURLOPT_SSLKEY]       = $keyCertPath;
+        $curlOptions[CURLOPT_SSLKEYPASSWD] = $keyCertPassword;
+        $curlOptions[CURLOPT_POST]         = 1;
+
+        $config->set('curl.options', $curlOptions);
+        $this->httpClient->setConfig($config);
+
+        // don't throw exceptions for 4xx errors
```

Comment:
Why are be suppressing errors?

Reply:
This is because Guzzle throws an error for a 400 status code. When that happens, we aren't able to get the ApplePayAuthorizeResponse we want, which should be returned no matter what the status code is.